python/cudf/cudf/core/udf/groupby_function.py

```diff
@@ -0,0 +1,851 @@
+    key = "GroupType.var"
+
+    def generic(self, args, kws):
+        return nb_signature(types.float64, recvr=self.this)
+
+
+class GroupIdxMax(AbstractTemplate):
+    key = "GroupType.idxmax"
+
+    def generic(self, args, kws):
+        return nb_signature(self.this.index_type, recvr=self.this)
+
+
+class GroupIdxMin(AbstractTemplate):
+    key = "GroupType.idxmin"
+
+    def generic(self, args, kws):
+        return nb_signature(self.this.index_type, recvr=self.this)
+
+
+@cuda_registry.register_attr
+class GroupAttr(AttributeTemplate):
+    key = GroupType
+
+    def resolve_max(self, mod):
+        return types.BoundFunction(
+            GroupMax, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_min(self, mod):
+        return types.BoundFunction(
+            GroupMin, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_size(self, mod):
+        return types.BoundFunction(
+            GroupSize, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_count(self, mod):
+        return types.BoundFunction(
+            GroupCount, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_sum(self, mod):
+        return types.BoundFunction(
+            GroupSum, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_mean(self, mod):
+        return types.BoundFunction(
+            GroupMean, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_std(self, mod):
+        return types.BoundFunction(
+            GroupStd, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_var(self, mod):
+        return types.BoundFunction(
+            GroupVar, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_idxmax(self, mod):
+        return types.BoundFunction(
+            GroupIdxMax, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+    def resolve_idxmin(self, mod):
+        return types.BoundFunction(
+            GroupIdxMin, GroupType(mod.group_scalar_type, mod.index_type)
+        )
+
+
+@cuda_lower("GroupType.max", GroupType(types.int32))
+@cuda_lower("GroupType.max", GroupType(types.int64))
+@cuda_lower("GroupType.max", GroupType(types.float64))
+def cuda_Group_max(context, builder, sig, args):
+    retty = sig.return_type
+
+    grp = cgutils.create_struct_proxy(sig.args[0])(
+        context, builder, value=args[0]
+    )
+    grp_type = sig.args[0]
+
+    group_dataty = grp_type.group_data_type
+    group_data_ptr = builder.alloca(grp.group_data.type)
+    builder.store(grp.group_data, group_data_ptr)
+
+    if grp_type.group_scalar_type == types.int32:
+        func = call_my_max_int32
+    elif grp_type.group_scalar_type == types.int64:
+        func = call_my_max_int64
+    elif grp_type.group_scalar_type == types.float64:
+        func = call_my_max_float64
+
+    result = context.compile_internal(
```

Comment:
Just return directly here, instead of assigning `result`.
```suggestion
    return context.compile_internal(
```